### ISSUE TITLE
Jekyll keeps using old version of CSS

### ISSUE THREAD

**Issue by rdgoite** [jekyll/docker#103]
Even after I completely remove the `_site` directory, Jekyll keeps using the outdated CSS. I'm not really sure where the container is getting the old values that seem to be cached somewhere. Also, whenever I check the CSS in `_site` directory, I'm prompted that the file is binary formatted. How do I force the container to use the latest CSS code?

**Comment by envygeeks** [comment]
The container is getting the folders from you.  We do not cache your folders (unless somehow volume started caching directories all of a sudden.  Which would make this an upstream problem... sadly.) Any data going into the container is provided by you and the system so if the system does not notify Docker properly or Docker does not notify us properly of a change then we are stuck with what we get.

Please provide more information because it doesn't sound to me like you are using native docker.

**Comment by rdgoite** [comment]
Yeah, which is why I'm baffled as to the source of this issue. The problem persists even after I completely remove the container, and rebuild it. Somehow the CSS that gets added into the `_site` directory is either outdated, or in some cases incomplete or malformed. There doesn't seem to be any mechanism for caching in place that can somehow go wrong.

Based on your response, I did some testing on Linux and the problem doesn't seem to be present in there. Could this have something to do with Docker environment on OS X?

**Comment by envygeeks** [comment]
@rdgoite there have been issues in the past where for some reason non-native Docker through Docker Machine and it's virtualization (through VirtualBox) didn't properly mount the file system leading to tons of problems.  I don't remember if they ever fixed that (or if it was fixed by VirtualBox itself.) I'll see if I can find those tickets as a few of them might have fixes.

**Comment by envygeeks** [comment]
@rdgoite sorry for not getting back to you faster, here are a few tickets:
https://www.virtualbox.org/ticket/9069
https://www.virtualbox.org/ticket/12597
https://www.virtualbox.org/ticket/14920

it seems this may be solved by https://github.com/brikis98/docker-osx-dev.  However, I know longer have an OS X computer laying around to play with so if you do try that docker-osx-dev and find it solves you problem it would be great if you update this ticket with your steps and then I (or you) can create a section in the README (or wiki) that will help guide users.

**Comment by envygeeks** [comment]
Oh yes, here is one of the original tickets: https://github.com/boot2docker/boot2docker/issues/652 they seem to be the same and related between docker-machine and boot2docker and there are dozens more, I just happened to see that one in my list of Docker bugs I've ran into that are pretty serious.

**Comment by rdgoite** [comment]
@envygeeks Thanks for the links. What I got from them is that most people addressed the server issues with virtualbox by turning off sendfile configuration (`sendfile off;`). I did the following: 

1. Define `.nginx/server.conf` in my Jekyll directory.
1. Added `sendfile off;` in `server.conf`

However, it didn't fix the issue. I even went as far as copying the original `jekyll.conf` in `/etc/nginx/conf.d`, modifying it by adding `sendfile off;` in both `server` and `location` sections, and copying this modified configuration into my own Jekyll image based on `jekyll/jekyll:pages`. 

I'd try the Docker OSX dev but I just got an invite to Docker for Mac beta, and I can confirm the problem does not exist there. Perhaps we can just wait for the new native Docker for Mac (and Windows) to be officially released.
